cartographer/field_types/schema_attribute.py

```diff
@@ -10,49 +10,98 @@ class SchemaAttribute(object):
     (string, int, bool, date, etc.)
     """
 
-    def __init__(self, model_attribute=None, is_property=True,
-                 serializer_method=None):
+    def __init__(self):
+        self.description_string = None
+        self.is_self_explanatory = False
+
+        self.model_property = None
+        self.model_method = None
+        self.serializer_method = None
+
+        self.is_nullable = False
+        self.is_optional_on_create = False
+        self.is_computed = False
+
+    def read_from(self, model_property=None, model_method=None, serializer_method=None):
         """
-        Note: only one of model_attribute and serializer_method should be provided.
-        Note: is_property is not relevant for serializer_method
-
-        :param model_attribute: the name of the attribute on the python model that should be (de)serialized
-        :param is_property: whether or not model_attribute is a property (if False, it is presumed it is a method)
-        :param serializer_method: the name of the method on the serializer object which uses this schema
-        which should be called to get the serialized value.
-        :return: an instance of SchemaAttribute,
-        which will later be used by a Parser or Serializer to move between JSON API and Python
+        Defines how to retrieve the value of the property. Only one parameter is allowed.
+
+        :param model_property: The property is read directly from the model using this property name
+        :param model_method: The property is read via a method on the model using this name
+        :param serializer_method: The property value is read via the serializer using a method of this name
         """
+        self.model_property = model_property
+        self.model_method = model_method
+        self.serializer_method = serializer_method
+        return self.sanity_check()
 
-        identifier_args = [model_attribute, serializer_method]
-        provided_identifiers = [identifier
-                                for identifier in identifier_args
-                                if identifier]
-        if len(provided_identifiers) > 1:
-            raise Exception("only one of [{}] should be provided".format(identifier_args.join(", ")))
+    def description(self, description_string):
+        """Set a description for this attribute. Useful for documentations"""
+        self.description_string = description_string
+        return self
 
-        self.model_attribute = model_attribute
-        self.is_property = is_property
-        self.serializer_method = serializer_method
+    def self_explanatory(self, is_self_explanatory=True):
+        self.is_self_explanatory = is_self_explanatory
```

Comment:
does it make sense for this to have an argument? `.self_explanatory(False)` feels unnecessary